### ISSUE TITLE
seed cannot work in the splitter.train_valid_test_split 

### ISSUE THREAD

**Issue by yxnyu** [deepchem/deepchem#3370]
## 🐛 Bug
------------------------------------------------------------------------
Hi, I just use the bbbp.csv with smiles and targets(p_np)
import pandas as pd
bbbp_dataset = dc.data.CSVLoader(tasks=["targets"], feature_field="smiles", featurizer = dc.feat.RawFeaturizer()).create_dataset("BBBP.csv")
splitter = dc.splits.ScaffoldSplitter()
train_dataset1, valid_dataset1, test_dataset1 = splitter.train_valid_test_split(bbbp_dataset, frac_train=0.8, frac_valid=0.1, frac_test=0.1, seed=1234)
train_dataset2, valid_dataset2, test_dataset2 = splitter.train_valid_test_split(bbbp_dataset, frac_train=0.8, frac_valid=0.1, frac_test=0.1, seed=11)

datasets = [train_dataset1, valid_dataset1, test_dataset1, train_dataset2, valid_dataset2, test_dataset2]
filenames = ['train_data1.csv', 'valid_data1.csv', 'test_data1.csv', 'train_data2.csv', 'valid_data2.csv', 'test_data2.csv']

for i, dataset in enumerate(datasets):
    df = dataset.to_dataframe()
    df.drop(columns=["X", "w"], inplace=True)
    df.to_csv(filenames[i], index=False)

df1 = pd.read_csv('train_data1.csv')
df2 = pd.read_csv('train_data2.csv')

if df1.equals(df2):
    print("The two CSV files have the same content.")
else:
    print("The two CSV files have different content.")
------------------------------------------------------------------------------------------------
after running this simple script, I got the _The two CSV files have the same content_. it seens that the seed cannot be used to split the dataset.
I am wondering how can I do？

**Comment by chertianser** [comment]
`ScaffoldSplit` is [designed to be deterministic](https://github.com/deepchem/deepchem/pull/3046/commits/62137e824098c193e8f5be8f9433c1876d481a26#r962226139) within deepchem. [Within the ScaffoldSplitter,](https://github.com/deepchem/deepchem/blob/0bc3139bb99ae7700ba2325a6756e33b6c327842/deepchem/splits/splitters.py#L1565) in `generate_scaffolds`, the final set of scaffolds are sorted in descending sizes of sets. The training, validation and test sets are then populated from the set of scaffolds. The `Splitter` class always takes `seed` as an optional argument, so it would pass correctly to `ScaffoldSplitter`, but the seed is not used at all in the splitter. Even if the dataset is shuffled, the sorting at the end of the splitting will remove any randomization.
